Support analysis / types with Rust Analyzer

Hofer-Julian opened the pull request:
Fixes #881
It still takes a bit until diagnostics arrive, but this way error messages from `cargo` are included as well.

Also moves the target dir to a common place, similar to how we do it when we compile Rust.

Hofer-Julian committed:
Support analysis / types with Rust Analyzer

Fixes #881
It still takes a bit until diagnostics arrive, but this way error messages from `cargo` are included as well.

sonnyp reviewed (commented):
I'm still only getting syntax diagnostics.

Anything specific I need to do?

![image](https://github.com/workbenchdev/Workbench/assets/19673/469d1e13-182b-4fa9-8d1d-2c4ec0274384)

![image](https://github.com/workbenchdev/Workbench/assets/19673/35bc730b-0667-4e96-b1a2-87d2fc0b8e3d)

Hofer-Julian commented:
> I'm still only getting syntax diagnostics.
> 
> Anything specific I need to do?
> 
> ![image](https://private-user-images.githubusercontent.com/19673/304914538-469d1e13-182b-4fa9-8d1d-2c4ec0274384.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5ODYxMjMsIm5iZiI6MTcwNzk4NTgyMywicGF0aCI6Ii8xOTY3My8zMDQ5MTQ1MzgtNDY5ZDFlMTMtMTgyYi00ZmE5LThkMWQtMmM0ZWMwMjc0Mzg0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE1VDA4MzAyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFlY2M2YjZiYzU3ZDIwMDk5MGI5MDVmZDA4YjExZmRiODA4YTBmMDk0ZDI5MWMwNTkzNDNkYjhhMzU4NmQ3MDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.J2BObaac9OpWobaQSHHrsIBa3yYfK8fW8iO6Xa78qqM)
> 
> ![image](https://private-user-images.githubusercontent.com/19673/304914597-35bc730b-0667-4e96-b1a2-87d2fc0b8e3d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5ODYxMjMsIm5iZiI6MTcwNzk4NTgyMywicGF0aCI6Ii8xOTY3My8zMDQ5MTQ1OTctMzViYzczMGItMDY2Ny00ZTk2LWIxYTItODdkMmZjMGI4ZTNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE1VDA4MzAyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc4NjAwZDUxOTUwMzRkMzY2Y2YzZmVlY2E0ZDRmYmJkMjJhNjA2ZGIxOGUzMmMwNWJjNDRiNTJlYmJlMWEwZWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.dUoHDsxMaXEHnHerrHPEgkBkKxYGiEsqiFfgd1y_wJE)

I shouldn't be necessary to do anything special, but I also noticed that it doesn't always work on my machine.

Before merging, we should also look out if this PR introduces regressions. If yes, maybe we just merge a subset of it.

theCapypara committed:
Python Previewer: Initialize Shumate (#884)

See:
- https://github.com/workbenchdev/demos/pull/94
- https://github.com/workbenchdev/Workbench/pull/879
- https://github.com/workbenchdev/Workbench/issues/738

Commit:
String fixes (#888)

sonnyp committed:
Use async enumerate_children (#889)

Diego-Ivan committed:
flatpak: Remove `/app/include` from cleanup (#886)

Before this change, Workbench installed via Flatpak was not able to run
Vala demos that required Shumate, Libportal or Libspelling because their
header files were removed by the cleanup process. The versions of
Workbench that are run from Builder or flatpak-vscode can actually
compile and execute this demos because they don't do any clean-up, and
therefore none of the header files are deleted, which is probably why
this issue went unnoticed for quite some time even when porting the
affected demos to Vala.

In order to test this changes and compare them with the current version
of Workbench, it is necessary to export a bundle and install it, so the
cleanup process is actually done.

Closes #885

Commit:
Add Bart Gravendeel to contributors (#890)

sonnyp committed:
Update modules (#891)

sonnyp committed:
Open Workbench windows maximized

sonnyp committed:
Fix extensions detection (#892)

sonnyp committed:
Solve brwap dangling process (#893)

Fixes https://github.com/workbenchdev/Workbench/issues/828

Hofer-Julian committed:
docs: Extend contributing section (#895)

`libxmljs` one of our nodejs dependencies requires a C++ compiler

sonnyp committed:
46 beta (#896)

sonnyp committed:
Fix appstream to make new Flathub validation pass

BharatAtbrat committed:
about: Add as contributor (#898)

sonnyp committed:
Use Flathub beta

janvhs committed:
Add pylsp as the python LSP (#857)

Co-authored-by: Sonny Piers <sonny@fastmail.net>
Co-authored-by: Marco Capypara Köpcke <hello@capypara.de>

sonnyp committed:
Use an AdwDialog for Extensions (#900)

theCapypara committed:
Replace Black with Ruff, add LSP plugin for Ruff (#902)

Co-authored-by: Sonny Piers <sonny@fastmail.net>

sonnyp committed:
cli: Add proper Python support (#903)

janvhs committed:
Add myself to to the about section (#904)

This commit adds my name to the contributor list for #857 

Thanks so much for the patience and help I received from you @sonnyp
@theCapypara

sonnyp committed:
Adapt to demos moving to src

sonnyp committed:
Update dependencies and release notes (#905)

Rirusha committed:
Add Rirusha to about.js (#907)

https://github.com/workbenchdev/demos/pull/120

sonnyp committed:
cli: Ignore Gtk.TreeIter deprecation in Text Fields

sonnyp committed:
Do not build separate locales

We only support English

sonnyp committed:
pre 46 maintenance (#909)

sonnyp committed:
46

sonnyp committed:
Update VTE 0.76 (#911)

sonnyp committed:
Clear own logs on start

sonnyp committed:
rust: Enable GNOME 46 (#910)

sonnyp committed:
Update Blueprint to 0.12.0
